src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -724,9 +723,16 @@ internal void DownloadAndInstall(PackageDownloadHandle packageDownloadHandle, st
                         {
                             var p = Package.FromDirectory(dynPkg.RootDirectory, DynamoViewModel.Model.Logger);
 
-                            pmExtension.PackageLoader.LoadPackages(new List<Package> {p});
-
-                            packageDownloadHandle.DownloadState = PackageDownloadHandle.State.Installed;
+                            if (p != null)
+                            {
+                                pmExtension.PackageLoader.LoadPackages(new List<Package> { p });
```

Comment:
This fix is on the Client side and we show a message that the downloaded package is Invalid instead of showing a message regarding the null reference exception.

Reply:
By the time Package.FromDirectory function is called, the package is already downloaded locally. In this method, it first looks for the pkg.json file in the root directory. If it doesn't find it, an exception is thrown. Then it tries to create a package object from the json header file. If any of this fails or if no pkg could be created from the json header file, then pkg would be null. From this, I think we can say that it is an invalid package.